### PR TITLE
Add a project contributor guide

### PR DESCRIPTION
**Pull request by per1234**
### Motivation

Documentation of how to contribute to the project gives everyone the opportunity to participate, while also reducing the maintenance effort and increasing the quality of contributions.

Currently, the ways and processes for contributing to this project are not fully documented, and the existing documentation is somewhat disorganized

### Change description

Add a guide documenting the various ways of contributing to the project.

Take advantage of [GitHub's "contributing guidelines" feature](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) to increase the visibility of the information to prospective contributors.

### Other information

See the rendered content here:

https://github.com/per1234/arduino-ide/blob/contributor-guide/docs/CONTRIBUTING.md

### Reviewer checklist

* [ ] PR addresses a single concern.
* [ ] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-ide/pulls) before creating one)
* [ ] PR title and description are properly filled.
* [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Comment by nmzaheer** [comment]
Great work @per1234 . However, I was thinking whether such documentation should live along with the code.

Why don't we shift this content to the repository's wiki. A great example of a wiki I've seen is of [VS Code] (https://github.com/microsoft/vscode/wiki). The sidebar on the right is clearly demarcated and very easy to navigate.

Another aspect that is not captured in the documentation is the Roadmap for Arduino IDE along with a Release philosophy. I think it is very important to lay down the roadmap along with a set of criteria that would lead to the stable release of v2. I believe there is some more work to be done in a few components before a stable release can be brought out. A roadmap stating the activities/ tasks that need to be completed should be put out in the public domain which would focus all our efforts on the same goal.

**Comment by per1234** [comment]
Thanks for your suggestions @nmzaheer.

> Why don't we shift this content to the repository's wiki.

I have extensive experience with GitHub Wikis and they are far inferior to using a repository in almost every way. You lose the incredible benefits of the pull request system. Even though a wiki is a Git repository under the hood, it is a strange repository and I don't think anyone ever works in them via a Git clone. So you miss the ability to edit multiple files at once, to work directly on the files with a professional quality text editor, to write proper commits.

The one thing of value to GitHub wikis is the ability to allow open editing of the content by anyone. That would not be appropriate with this content. Anyone may propose changes to it, but that proposal must pass through a review process by the maintainers.

>The sidebar on the right is clearly demarcated and very easy to navigate.

If we wanted some navigation, we would set up a MkDocs-based website as we have done with other tooling projects:

- https://arduino.github.io/arduino-cli/latest/
- https://arduino.github.io/arduino-lint/latest/
- https://arduino.github.io/arduino-fwuploader/latest/

> lay down the roadmap along with a set of criteria that would lead to the stable release of v2. I believe there is some more work to be done in a few components before a stable release can be brought out. A roadmap stating the activities/ tasks that need to be completed should be put out in the public domain which would focus all our efforts on the same goal.

GitHub has a nice system for that:

https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones

Adding milestones to issues is much easier than maintaining a bespoke list that would actually be inferior in functionality due to not having the searching, sorting, and categorization features of GitHub Issues.

**Comment by nmzaheer** [comment]
> If we wanted some navigation, we would set up a MkDocs-based website as we have done with other tooling projects
I think this would be a better option as there would be consistency across projects then. Let others also weigh in on this.

> Adding milestones to issues is much easier than maintaining a bespoke list that would actually be inferior in functionality due to not having the searching, sorting, and categorization features of GitHub Issues.

+1 for this too. As long as there is a way to track our progress, it would be great.

File: docs/contributor-guide/beta-testing.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+![tester build artifacts](assets/tester-build-artifacts.png)
+
+#### Notes for macOS
+
+Beginning in macOS 10.14.5, the software [must be notarized to run](https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution).
+
+For security reasons, signing and notarization are disabled when creating tester builds for pull requests from forks of this repository. This means that macOS will block you from running the tester builds for those PRs.
+
+Due to this limitation, Mac users have two options for testing PRs from forks:
```
Comment (kittaakos) [inline docs/contributor-guide/beta-testing.md:52]: There is a third one, the user-friendly one:
 - Open `System Preferences` > `Security & Privacy` > `General` > Allow IDE2 to run in the `Allow apps downloaded...` section. You will be prompted for your password.

**Review by kittaakos — APPROVED** [review]
Excellent work, Per 👍

File: docs/contributor-guide/beta-testing.md (hunk 1 above)
Comment (per1234) [reply docs/contributor-guide/beta-testing.md:52]: Thanks @kittaakos! I think that is a much better approach.

I decided to replace the current "risky approach" instructions with the ones you suggested. Please let me know what you think:

https://github.com/arduino/arduino-ide/pull/1419/commits/eb6fe4f3b720af7acb86014c2a7acc7a2cc088bc

https://github.com/per1234/arduino-ide/blob/contributor-guide/docs/contributor-guide/beta-testing.md#the-risky-approach

Comment (kittaakos) [reply docs/contributor-guide/beta-testing.md:52]: Perfect 👍 

I am resolving this thread.

**Pull request merged**